apps/sms/views/inbox/js/inbox.js

```diff
@@ -140,6 +147,10 @@ var InboxView = {
     if (args.notifyAboutSavedDraft) {
       this.showDraftSavedBanner();
     }
+
+    if (!args.meta.prev) {
+      this.renderThreads();
+    }
```

Comment:
Well, I'm happy to have better fix! But to give more meaningful alternatives, could you please tell me what exactly feels wrong in the current solution to be on the same page?

My current solution was based on two assumptions:

* ```Navigation.once('navigated')``` in startup.js is a more or less long-living hack that will go away - start rendering treads in another panel when "some" undetermined other panel is loaded.
* ```Navigation.ensureCurrentPanel``` is hack either for the old notification API that doesn't correspond to the spec (notification on close action), that also will go away.

So I just wanted to keep all those hacks together, but anyway, I agree it's a hack :)

> force that renderThreads runs only once -- if (!renderingOrRendered) { this.renderThreads() }

This will make file with what we're going to live with in the future more complex for nothing. So I'd rather not do it.

> do not call it here at all (for now) -- use apps/sms/views/conversation/js/startup.js instead

I suspect you meant ```apps/sms/views/inbox/js/startup.js``` and ```apps/sms/views/shared/js/startup.js```? If so, yeah we can call ```renderThreads``` directly in startup files for now if we can't figure out better alternatives.